monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -311,8 +311,8 @@ sealed abstract class Task[+A] extends Serializable { self =>
     *         that can be used to extract the result or to cancel
     *         a running task.
     */
-  def runAsync(implicit s: Scheduler): CancelableFuture[A] =
-    TaskRunLoop.startAsFuture(this, s)
+  def runAsync(implicit s: Scheduler, opts: Options): CancelableFuture[A] =
```

Comment:
@alexandru just a note on this. I can only define a default value for only one overloaded runAsync function. If I define it on this one, the other one can't have it and that is why I chose the implicit default options val because it broke the code in many places for the runAsync with the callback function.

I can make that change but it will affect code in reactive module, so the impact is going to be bigger.

Right now the default value is defined in the first runAsync function:
``` scala
def runAsync(cb: Callback[A])(implicit s: Scheduler, opts: Options = defaultOptions): Cancelable
```